app/repositories/AdminDataRepository.scala

```diff
@@ -17,101 +17,58 @@
 package repositories
 
 import com.google.inject.Inject
+import com.mongodb.client.model.FindOneAndUpdateOptions
 import models.FeatureToggle
-import org.slf4j.{Logger, LoggerFactory}
+import org.mongodb.scala.model.{Filters, IndexModel, IndexOptions, Indexes}
+import org.mongodb.scala.model.Updates.set
 import play.api.libs.json._
-import play.api.Configuration
-import play.modules.reactivemongo.ReactiveMongoComponent
-import reactivemongo.api.commands.WriteResult
-import reactivemongo.api.indexes.{Index, IndexType}
-import reactivemongo.bson.{BSONDocument, BSONObjectID}
-import reactivemongo.play.json.ImplicitBSONHandlers._
-import uk.gov.hmrc.mongo.ReactiveRepository
+import play.api.{Configuration, Logging}
+import repositories.FeatureToggleMongoFormatter.{FeatureToggles, featureToggles, id}
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.{Codecs, PlayMongoRepository}
 
 import scala.concurrent.{ExecutionContext, Future}
 
+object FeatureToggleMongoFormatter {
+  case class FeatureToggles(_id: String, toggles: Seq[FeatureToggle])
+
+  implicit val featureToggleMongoFormatter: Format[FeatureToggles] =  Json.format[FeatureToggles]
+
+  val id = "_id"
+  val featureToggles = "toggles"
+}
+
 class AdminDataRepository @Inject()(
-                                     mongoComponent: ReactiveMongoComponent,
+                                     mongoComponent: MongoComponent,
                                      configuration: Configuration
                                    )(implicit val ec: ExecutionContext)
-  extends ReactiveRepository[JsValue, BSONObjectID](
+  extends PlayMongoRepository[FeatureToggles](
     collectionName = configuration.get[String](path = "mongodb.pension-administrator-cache.admin-data.name"),
-    mongo = mongoComponent.mongoConnector.db,
-    domainFormat = implicitly
-  ) {
-
-  override val logger: Logger = LoggerFactory.getLogger("AdminDataRepository")
-
-  private val featureToggleDocumentId = "toggles"
-
-  val collectionIndexes = Seq(
-    Index(
-      key = Seq((featureToggleDocumentId, IndexType.Ascending)),
-      name = Some(featureToggleDocumentId),
-      background = true,
-      unique = true
+    mongoComponent = mongoComponent,
+    domainFormat = FeatureToggleMongoFormatter.featureToggleMongoFormatter,
+    indexes = Seq(
+      IndexModel(
+        Indexes.ascending("toggles"),
+        IndexOptions().name("toggles").unique(true).background(true))
```

Comment:
Should these be referencing the val from the `FeatureToggleMongoFormatter` object?
```
val featureToggles = "toggles"
```